### ISSUE TITLE
Add content warning system

### ISSUE THREAD

**Issue by rubenwardy** [minetest/serverlist#55]
Servers with certain content warnings should be hidden by default. A new query argument should be supported to show certain categories, for example:  `?show=all`

## Alternative

Ban problematic content in #54

**Comment by runsy** [comment]
Flags are ugly, best a compact system:
https://forum.minetest.net/viewtopic.php?f=5&t=25890&hilit=sammy+rating

**Comment by Warr1024** [comment]
I like the idea of having the server content system align largely with CDB's content warnings, especially since that means we can default the server's content flags to a union of the content flags from installed mods automatically, for server owners who forget to set them up manually.

I also like the idea of some generic "adults_only" kind of flag that's independent of more specific flags, which can be used by server owners to warn that children are discouraged or unwanted in certain places.

There might also be some value in having flags that are like the inverse of these, like "family_friendly" in the general sense, or "no_swearing" or "no_adult_discussion" for more specifics; people may want to use these to filter out servers in the server list where the kinds of things they may want to do would be disallowed.  Though I'm rather less sure of this because it complicates filtering (you can't just hide anything with any flag by default) and it would only be useful if players were given filtering options in the UI.

**Comment by rubenwardy** [comment]
sfan5 doesn't want to have to moderate or enforce content warnings, so is suggesting just having a single flag for "adult only" servers - which includes both servers that have mature content and those that don't want kids

**Comment by Mooncarguy** [comment]
I'd just suggest a "hidden" flag added by minetest.net moderation. Flagged servers are no longer shown on the serverlist. However newer minetest versions should include an opt-in feature to display those servers. However logically this isn't available for old MT versions, but there are so many players running around still having those versions installed, especially young ones, so it's worth it.

**Comment by runsy** [comment]
> sfan5 doesn't want to have to moderate or enforce content warnings, so is suggesting just having a single flag for "adult only" servers - which includes both servers that have mature content and those that don't want kids

It seems to me the most reasonable thing to do. A scarlet mark, let's call it.

It's OK to have servers for children, teenagers and also servers with X content for responsible and healthy adults.

It's time for Minetest to stop being less namby-pamby and get older.

**Comment by BluebirdGreycoat** [comment]
Let me come at this from a different angle: I vote for splitting the serverlist into categories, and allowing the client to select which category to view. 

Advantages:

1) servers with the potential to cause problems with public relations can have their own category, so they aren't seen right away by newcomers unless asked for.

2) servers with one style of game (example: survival) no longer have to compete rankings-wise with servers that have a completely different focus (e.g., creative).

Suggested categories:

```
all_except_adult
pvp
puzzle
creative
survival
undefined
adult
```

**Comment by runsy** [comment]
> Let me come at this from a different angle: I vote for splitting the serverlist into gamemode categories, and allowing the client to select which category to view.
> 
> Advantages:
> 
>     1. servers with the potential to cause problems with public relations can have their own category, so they aren't seen right away by newcomers unless asked for.
> 
>     2. servers with one style of game (example: survival) no longer have to compete rankings-wise with servers that have a completely different focus (e.g., creative).
> 
> 
> Suggested categories:
> 
> ```
> all_except_adult
> pvp
> puzzle
> creative
> survival
> undefined
> adult
> ```

No, adult category, should be hidden/disabled by default. And ONLY BE SET MANUALLY in minetest.conf file, so no accesible for children easily! Even it does not exist per se, but added manually.

**Comment by runsy** [comment]
Maybe a committee of wise men of various reputable people in Minetest for flagrant cases of abuse of non-use of this category, that are unlisted, not just one person.. It would be very strange. it should be noted that TODAY YOU CAN ALREADY CREATE AN ADULT SERVER of Minetest. It is feasible. And I believe that they do not exist or are not so obvious.

**Comment by Festus1965** [comment]
> sfan5 doesn't want to have to moderate or enforce content warnings, so is suggesting just having a single flag for "adult only" servers - which includes both servers that have mature content and those that don't want kids

maybe a kind of 'voting' from 'helpers' that take every day time for 1-2 servers and flag them - without a direct effect, but after maybe 5-10 certified voting / suggestions it goes active ?

It is similar to our guy found that server and posted a comment - quite easier to have a possibility to rate servers for a allowed group, as with 400, now just 387 servers now sure too much for one person, and than also to person related.

Short suggestion for content suggestion with 2 or 3 keywords from the chosen visitors, and every 15 days checked and set real if they are clear stated, like 4 of 5 say adult, or more specific blood-version ...

**Comment by Mooncarguy** [comment]
While I'm not a lawyer or legal expert and I have no clue of the Netherlands' law,

I'd like to suggest some proper research to be done on what content is legally required to be flagged or even entirely removed.

Not knowing of a regulation does not protect from fines or criminal proceedings and it would be very short-sighted not to consider the possibility of an accidental breach.
